packages/kokkos/cmake/kokkos_arch.cmake

```diff
@@ -155,7 +155,7 @@ IF (KOKKOS_ENABLE_CUDA_CONSTEXPR)
   ENDIF()
 ENDIF()
 
-IF (KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+IF (KOKKOS_CXX_COMPILER_ID STREQUAL Clang AND KOKKOS_ENABLE_CUDA)
   SET(CUDA_ARCH_FLAG "--cuda-gpu-arch")
   GLOBAL_APPEND(KOKKOS_CUDA_OPTIONS -x cuda)
   IF (KOKKOS_ENABLE_CUDA)
```

Comment:
@ndellingwood, given the above addition of `AND KOKKOS_ENABLE_CUDA`, can't you remove this `IF (KOKKOS_ENABLE_CUDA)`?  It would just simplify the code.

Reply:
Good point, I was hurrying to get this fix in the PR tester. Let me get @jjwilke feedback as well, he proposed a different change below and may want this removed though it seems like an appropriate safety check to add.

Reply:
@jjwilke and @ndellingwood, if the user is not asking for CUDA, why add **anything** for CUDA in CMake variables at all?  What good can come of that?  That seems like you are just asking for trouble.  If you have to add REMOVE statements, that is a sign of a fragile systems.